HARK/ConsumptionSaving/ConsAggShockModel.py

```diff
@@ -550,11 +550,326 @@ def getMrkvNow(self):  # This function exists to be overwritten in StickyE model
+
+        Returns
+        -------
+        None
+        '''
+        self.T_sim = Economy.act_T                   # Need to be able to track as many periods as economy runs
+        self.kInit = Economy.KSS                     # Initialize simulation assets to steady state
+        self.MrkvInit = Economy.MrkvNow_init         # Starting Markov state for the macroeconomy
+        self.Mgrid = Economy.MSS*self.MgridBase      # Aggregate market resources grid adjusted around SS capital ratio
+        self.AFunc = Economy.AFunc                   # Next period's aggregate savings function
+        self.DeprFac = Economy.DeprFac               # Rate of capital depreciation
+        self.CapShare = Economy.CapShare             # Capital's share of production
+        self.LbrInd = Economy.LbrInd                 # Idiosyncratic labor supply (when employed)
+        self.UrateB = Economy.UrateB                 # Unemployment rate in bad state
+        self.UrateG = Economy.UrateG                 # Unemployment rate in good state
+        self.ProdB = Economy.ProdB                   # Total factor productivity in bad state
+        self.ProdG = Economy.ProdG                   # Total factor productivity in good state
+        self.MrkvIndArray = Economy.MrkvIndArray     # Transition probabilities among discrete states
+        self.MrkvAggArray = Economy.MrkvArray        # Transition probabilities among aggregate discrete states
+        self.addToTimeInv('Mgrid', 'AFunc', 'DeprFac', 'CapShare', 'UrateB', 'LbrInd',\
+                          'UrateG', 'ProdB', 'ProdG', 'MrkvIndArray', 'MrkvAggArray')
+    
+    def makeGrid(self):
+        '''
+        Construct the attribute aXtraGrid from the primitive attributes aMin,
+        aMax, aCount, aNestFac.
+        '''
+        self.aGrid = makeGridExpMult(self.aMin,
+                                         self.aMax,
+                                         self.aCount,
+                                         self.aNestFac)
+        self.addToTimeInv('aGrid')
+        
+    def updateSolutionTerminal(self):
+        '''
+        Construct the trivial terminal period solution (initial guess).
+        '''
+        cFunc_terminal = 4*[IdentityFunction(n_dims=2)]
+        vPfunc_terminal = [MargValueFunc2D(cFunc_terminal[j], self.CRRA) for j in range(4)]
+        self.solution_terminal = ConsumerSolution(cFunc = cFunc_terminal,
+                                                  vPfunc = vPfunc_terminal)
+        
+    def preComputeArrays(self):
+        '''
+        Construct the attributes X, which will be used by the one period solver.
+        '''
+        # Get array sizes
+        aCount = self.aGrid.size
+        Mcount = self.Mgrid.size
+        
+        # Make tiled array of end-of-period idiosyncratic assets (order: a, M, s, s')
+        aNow_tiled = np.tile(np.reshape(self.aGrid, [aCount, 1, 1, 1]), [1, Mcount, 4, 4]) # shape (aCount, Mcount, 4, 4)
```

Comment:
In general, I think the code formatting of this PR is really nice.
I was going to ask if you used `black` for it?

I'm surprised by this long line (in terms of number of characters) though.

Reply:
No, I didn't use black.  All of my comments with the shape of arrays can be removed (will do that now).  I put those in while explaining the code the Mridul.